p2p/src/interface/types.rs

```diff
@@ -13,12 +13,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#[derive(Debug, serde::Serialize)]
+use serde::Serialize;
+
+use crate::types::peer_id::PeerId;
+
+#[derive(Debug, Serialize)]
 pub struct ConnectedPeer {
-    // TODO: Replace String with actual type, once libp2p removed
-    pub peer_id: String,
+    pub peer_id: PeerId,
 
-    // TODO: Replace String with actual type, once libp2p removed
```

Comment:
Are there chances that we can replace this string with the actual type too?

Reply:
The problem is that we don't know the actual type.
For example here:
```
impl<T: ToString> From<&PeerContext<T>> for ConnectedPeer {
    fn from(context: &PeerContext<T>) -> Self {
        ConnectedPeer {
            peer_id: context.info.peer_id,
            address: context.address.to_string(),
            inbound: context.role == Role::Inbound,
            ban_score: context.score,
        }
    }
}
```

I suppose we could try to replace it with the `PeerAddress` type, which can be constructed from a generic network address (`PeerContext` would need to implement `serde::Serialize` then).